project_recalculate/models/project_task.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2015 Antonio Espinosa
+# Copyright 2015 Endika Iglesias
+# Copyright 2015 Javier Espìnosa
+# Copyright 2017 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from pytz import timezone, utc
+
+from odoo import exceptions
+from odoo import models, fields, api, _
+from datetime import datetime
+
+
+class ProjectTask(models.Model):
+    _inherit = 'project.task'
+
+    from_days = fields.Integer(
+        string='From days',
+        help='Anticipation days from date begin or date end', default=0)
+    estimated_days = fields.Integer(
+        string='Estimated days', help='Estimated days to end', default=1,
+        oldname='anticipation_days')
+    include_in_recalculate = fields.Boolean(
+        related="stage_id.include_in_recalculate", readonly=True,
+    )
+
+    @api.constrains('estimated_days')
+    def _estimated_days_check(self):
+        for task in self:
+            if task.estimated_days <= 0:
+                raise exceptions.ValidationError(
+                    _('Estimated days must be greater than 0.')
+                )
+
+    def _dates_onchange(self, vals):
```

Comment:
I don't like this method name. It's not an onchange, so it's confusing. Could you change it?